rpc/src/test/scala/org/bitcoins/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,517 @@
+      logger.info(count.toString)
+      assert(count == 0)
+    }
+  }
+
+  it should "be able to get the best block hash" in {
+    val bestHashF = client.getBestBlockHash
+    bestHashF.map { hash =>
+      logger.info(hash.toString)
+      assert(true)
+    }
+  }
+
+  it should "be able to get the mining info" in {
+    val miningInfoF = client.getMiningInfo
+    miningInfoF.map { info =>
+      logger.info(info.toString)
+      assert(info.chain == "regtest")
+    }
+  }
+
+  it should "be able to get the chain tips" in {
+    val chainTipsF = client.getChainTips
+    chainTipsF.map { tipArray =>
+      tipArray.foreach { tip =>
+        logger.info(tip.toString)
+        assert(tip.status == "active")
+      }
+      assert(true)
+    }
+  }
+
+  it should "be able to get the network info" in {
+    val networkInfoF = client.getNetworkInfo
+    networkInfoF.map { info =>
+      logger.info(info.toString)
+      assert(info.networkactive)
+      assert(info.connections == 0)
+      assert(info.localrelay)
+    }
+  }
+
+  it should "be able to generate blocks" in {
+    val blocksF = client.generate(3)
+    blocksF.map { blocks =>
+      blocks.foreach(block => logger.info(block.toString))
+      assert(blocks.length == 3)
+    }
+  }
+
+  it should "be able to generate blocks and then get their serialized headers" in {
+    val blocksF = client.generate(2)
+    blocksF.flatMap { blocks =>
+      blocks.foreach(block => logger.info(block.toString))
+      val headerF1 = client.getBlockHeaderRaw(blocks(1))
+      headerF1.map { header =>
+        logger.info(header.toString)
+        assert(header.previousBlockHashBE == blocks(0))
+      }
+    }
+  }
+
+  it should "be able to generate blocks and then get their headers" in {
+    val blocksF = client.generate(2)
+    blocksF.flatMap { blocks =>
+      val headerF0 = client.getBlockHeader(blocks(0))
+      headerF0.map { header =>
+        logger.info(header.toString)
+        assert(header.nextblockhash.contains(blocks(1)))
+      }
+      val headerF1 = client.getBlockHeader(blocks(1))
+      headerF1.map { header =>
+        logger.info(header.toString)
+        assert(header.previousblockhash.contains(blocks(0)))
+        assert(header.nextblockhash.isEmpty)
+      }
+    }
+  }
+
+  it should "be able to get the balance" in {
+    val balanceF = client.getBalance
+    balanceF.flatMap { balance =>
+      logger.info(balance.toString)
+      assert(balance.toBigDecimal > 0)
+      client.generate(1).flatMap { _ =>
+        client.getBalance.map { newBalance =>
+          assert(balance.toBigDecimal < newBalance.toBigDecimal)
+        }
+      }
+    }
+  }
+
+  it should "be able to get block hash by height" in {
+    client.generate(2).flatMap { blocks =>
+      client.getBlockCount.flatMap { count =>
+        client.getBlockHash(count).map { hash =>
+          assert(blocks(1) == hash)
+        }
+        client.getBlockHash(count - 1).map { hash =>
```

Comment:
Note that this Future might be finished running before 

```
assert(blocks(1) == hash)
```

Is done running, which means that that assertion will not get tested.

Reply:
Confirmed that this is the case yesterday. This needs to be re-written